File: cmd_users_admin.py
# cmd_users_admin.py
from db_manager import users_col

# ======= إعداد المطورين =======
DEV_IDS = [5860391324, 7076215547, 7855813063]  # أضف كل المطورين هنا
COMMANDS = ["ادارة_المستخدمين", "المستخدمين"]
MAX_MSG_LENGTH = 4000  # الحد الأقصى لرسائل Telegram

async def handle(bot, message):
    uid = message.from_user.id
    text = message.text.strip()

    if text not in COMMANDS:
        return

    if uid not in DEV_IDS:
        await bot.send_message(message.chat.id, "❌ هذا الأمر للمطورين فقط")
        return

    users = await users_col.find({}).to_list(length=None)
    total = len(users)

    banned = []
    active = []

    for user in users:
        user_id = user["uid"]
        gold = user.get("gold", 0)
        bank = user.get("bank", 0)
        is_banned = user.get("banned", False)
        name = user.get("name", f"مستخدم {user_id}")

        info = f"- {name} | ID: {user_id} | 💰 {gold} | 🏦 {bank}"
        if is_banned:
            banned.append(info + " | 🚫 محظور")
        else:
            active.append(info)

    text_header = f"""╔═════════════════╗
  أهلاً بك يا {message.from_user.first_name} في إدارة المستخدمين
╚═════════════════╝

👥 عدد المستخدمين الكلي: {total}

🚫 المحظورين:
"""

    text_banned = "\n".join(banned)
    text_active = "\n\n✅ غير المحظورين:\n" + "\n".join(active)

    full_text = text_header + text_banned + text_active

    # تقسيم الرسالة لو تجاوزت الحد
    for i in range(0, len(full_text), MAX_MSG_LENGTH):
        await bot.send_message(message.chat.id, full_text[i:i+MAX_MSG_LENGTH])

File: cmd_users_admin.js
// cmd_users_admin.js
const db_manager = require('./db_manager');

const COMMANDS = ["ادارة_المستخدمين", "المستخدمين"];
const MAX_MSG_LENGTH = 4000; // الحد الأقصى لرسائل Telegram

// قائمة المطورين المصرح لهم
const DEV_IDS = [5860391324, 123456789, 987654321]; // ضيف كل المطورين هنا

async function handle(ctx) {
    const uid = ctx.from.id;
    const text = ctx.message.text.trim();

    if (!COMMANDS.includes(text)) return;

    if (!DEV_IDS.includes(uid)) {
        return ctx.reply("❌ هذا الأمر للمطورين فقط");
    }

    const users = await db_manager.users.find({}).toArray();
    const total = users.length;

    const banned = [];
    const active = [];

    for (const user of users) {
        const user_id = user.uid;
        const gold = user.gold || 0;
        const bank = user.bank || 0;
        const is_banned = user.banned || false;

        const info = `- ID: ${user_id} | 💰 ${gold} | 🏦 ${bank}`;
        if (is_banned) banned.push(info + " | 🚫 محظور");
        else active.push(info);
    }

    const text_header = 
`╔═════════════════╗
  أهلاً بك يا ${ctx.from.first_name} في إدارة المستخدمين
╚═════════════════╝

👥 عدد المستخدمين الكلي: ${total}

🚫 المحظورين:
`;

    const text_banned = banned.length ? banned.join("\n") : "— لا يوجد —";
    const text_active = "\n\n✅ غير المحظورين:\n" + (active.length ? active.join("\n") : "— لا يوجد —");

    const full_text = text_header + text_banned + text_active;

    // تقسيم الرسالة لو تجاوزت الحد
    for (let i = 0; i < full_text.length; i += MAX_MSG_LENGTH) {
        await ctx.reply(full_text.slice(i, i + MAX_MSG_LENGTH));
    }
}

module.exports = { handle };
